[TRTLLM-5171] chore: Remove GptSession/V1 from TRT workflow

Funatiq opened the pull request:
## Description

Please explain the issue and the solution in short.

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

Funatiq committed:
chore: Remove GptSession/V1 from TRT workflow

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove stateful decoders

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove GptSession buffers

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove GptSession utils

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove GptSession kernels

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove V1 GPT models from tests

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

Funatiq committed:
chore: Remove gptSessionBenchmark from scripts and docs

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove gptSession IO classes

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #4241](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4241/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4241](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4241/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3030](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3030/) completed with status: 'FAILURE'

Funatiq committed:
chore: Remove GptSession from test lists

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq committed:
chore: Remove GptSession from docs

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --stage-list "H100_PCIe-CPP-1"

tensorrt-cicd commented:
[PR_Github #4259](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4259/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4259](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4259/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3046 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3046/) completed with status: 'FAILURE'

Funatiq committed:
chore: Remove useless encoder test

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run --stage-list "H100_PCIe-CPP-1"

tensorrt-cicd commented:
[PR_Github #4352](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4352/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3121 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3121/) completed with status: 'SUCCESS'

jdebache reviewed (commented):
Looks good overall. Two small comments / questions:

- Are we confident that we are not losing too much incidental testing by removing the `gptSession` tests?
- I've encountered many symbols only referenced in the `gptSession` path. Are you sure you removed all of those, as they should be now unused.

DomBrown commented:
LGTM. Nice. Julien's questions are good ones, though.

Funatiq committed:
chore: Remove mActualBatchSize from DecoderState

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #4869](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4869/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4869](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4869/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3527](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3527/) completed with status: 'FAILURE'

Funatiq committed:
chore: Remove static batching from ExecutorTest

- Updated `validateContextLogits` and `validateGenerationLogits` functions to remove the `batchingType` parameter.
- Adjusted related test functions to reflect the changes in parameter lists.
- Cleaned up the instantiation of test cases to eliminate unnecessary batchingType references.

Signed-off-by: Robin Kobus <19427718+Funatiq@users.noreply.github.com>

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #4906](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4906/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4906](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4906/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3558](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3558/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #5001](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5001/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #5001](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5001/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3634](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3634/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #5020](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5020/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #5020](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5020/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3652](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3652/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #5044](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5044/) [ run ] triggered by Bot

Funatiq commented:
> Are we confident that we are not losing too much incidental testing by removing the `gptSession` tests?

Yes, the models are also covered in `TrtModelRealDecoderTest` and `ExecutorTest`. The only models that we lose are those that do not support IFB anyway.

> I've encountered many symbols only referenced in the `gptSession` path. Are you sure you removed all of those, as they should be now unused.

I cleaned up everything I could find.

tensorrt-cicd commented:
[PR_Github #5044](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5044/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3672](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3672/) completed with status: 'FAILURE'

Funatiq commented:
/bot run

tensorrt-cicd commented:
[PR_Github #5149](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5149/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #5149](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5149/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3755](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3755/) completed with status: 'FAILURE'

Funatiq commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #5193](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5193/) [ run ] triggered by Bot
